mapper-runtime/src/main/java/com/datastax/oss/driver/internal/mapper/DaoBase.java

```diff
@@ -114,35 +118,105 @@ protected static SimpleStatement replaceKeyspaceAndTablePlaceholders(
     return SimpleStatement.newInstance(queryString);
   }
 
-  public static BoundStatementBuilder populateBoundStatementWithAttributes(
+  public static BoundStatementBuilder populateBoundStatementWithStatementAttributes(
       BoundStatementBuilder builder, StatementAttributes attributes) {
 
-    builder =
-        builder
-            .setExecutionProfileName(attributes.getExecutionProfileName())
-            .setExecutionProfile(attributes.getExecutionProfile())
-            .setConsistencyLevel(attributes.getConsistencyLevel())
-            .setSerialConsistencyLevel(attributes.getSerialConsistencyLevel())
-            .setIdempotence(attributes.isIdempotent())
-            .setPageSize(attributes.getPageSize())
-            .setRoutingKey(attributes.getRoutingKey())
-            .setRoutingKeyspace(attributes.getRoutingKeyspace())
-            .setPagingState(attributes.getPagingState())
-            .setTimeout(attributes.getTimeout())
-            .setQueryTimestamp(attributes.getQueryTimestamp())
-            .setNode(attributes.getNode())
-            .setRoutingToken(attributes.getRoutingToken());
-    for (String customPayloadKey : attributes.getCustomPayload().keySet()) {
-      builder =
-          builder.addCustomPayload(
-              customPayloadKey, attributes.getCustomPayload().get(customPayloadKey));
+    if (attributes.getExecutionProfileName().isPresent()) {
+      builder = builder.setExecutionProfileName(attributes.getExecutionProfileName().get());
+    }
```

Comment:
You can simplify this as:
```java
attributes.getExecutionProfileName().ifPresent(builder::setExecutionProfileName);
```
You just need to make `builder` final (for the custom payload loop you don't need to reassign builder, we know it's mutable).